EESSI-pilot-install-software.sh

```diff
@@ -107,22 +110,34 @@ echo ">> Setting up \$MODULEPATH..."
 module --force purge
 # ignore current $MODULEPATH entirely
 module unuse $MODULEPATH
-module use $EASYBUILD_INSTALLPATH/modules/all
+if [[ -z "$EESSI_REBUILD_PATH" ]]; then
+    # Default behaviour is to do a full build
+    module use $EASYBUILD_INSTALLPATH/modules/all
+else
+    # If EESSI_REBUILD_PATH is set, we assume software is built and we are just rebuilding the module tree
+    export EASYBUILD_INSTALLPATH_MODULES=$EESSI_REBUILD_PATH/modules
+    [[ -z "$EESSI_REBUILD_MODULEPATH" ]] && module use $EESSI_REBUILD_PATH/modules/all || module use $EESSI_REBUILD_MODULEPATH
+    # Use preferred MNS
+    [[ -z "$EESSI_REBUILD_MNS" ]] && export EASYBUILD_MODULE_NAMING_SCHEME=EasyBuildMNS || export EASYBUILD_MODULE_NAMING_SCHEME=$EESSI_REBUILD_MNS
+    # Only rebuild modules
+    export EASYBUILD_MODULE_ONLY=1
+    # Make sure we can write our lock files
+    export EASYBUILD_LOCKS_DIR=$EESSI_REBUILD_PATH/software
+fi
+
 if [[ -z ${MODULEPATH} ]]; then
     fatal_error "Failed to set up \$MODULEPATH?!"
 else
     echo_green ">> MODULEPATH set up: ${MODULEPATH}"
 fi
 
-REQ_EB_VERSION='4.4.1'
-
-echo ">> Checking for EasyBuild module..."
-ml_av_easybuild_out=$TMPDIR/ml_av_easybuild.out
-module avail 2>&1 | grep -i easybuild/${REQ_EB_VERSION} &> ${ml_av_easybuild_out}
+echo ">> Checking for ${EB_MODULE_NAME} module (required version is ${REQ_EB_VERSION})..."
+ml_show_easybuild_out=$TMPDIR/ml_show_easybuild.out
+module show ${EB_MODULE_NAME}/${REQ_EB_VERSION} &> ${ml_show_easybuild_out}
```

Comment:
@boegel I see you used grep to get around https://github.com/EESSI/software-layer/issues/89, this approach also works, not sure which  is better...